motto/agents/openrouter_agent.py

```diff
@@ -0,0 +1,76 @@
+from .metta_agent import Agent
+import os
+import requests
+import json
+
+key = os.environ.get('OPENROUTER_API_KEY')
+
+class Function:
+    def __init__(self, name, arguments):
+        self.name = name
+        self.arguments = arguments
+
+
+class ToolCalls:
+    def __init__(self, tool_id, call_type, function):
+        self.id = tool_id
+        self.type = call_type
+        self.function = function
+
+
+class MessageClass:
+    def __init__(self, role, content, tool_calls=None):
+        self.role = role
+        self.content = content
+        self.tool_calls = tool_calls
+
+
+class OpenRouterAgent(Agent):
+    '''
+    OpenRouter is a unified interface for LLMs, this agent use OpenRouter API to get responses from LLM
+    '''
+
+    def __init__(self, model="openai/gpt-3.5-turbo", stream=False):
+        self._model = model
+        self.stream_response = stream
+
+    def __call__(self, messages, functions=[]):
+        if key is None:
+            return MessageClass("system", "No OPENROUTER_API_KEY is provided")
```

Comment:
Do we need a special class for this?

Reply:
I think yes, It was made to return responses in the same format as openai responses.  And do not add additional cases in llm_gate